compile error: call of overloaded ‘abs(unsigned int)’ is ambiguous

After applying fix for #49, still getting compilation errors in `sample_hevc_fei`.

Compiler version is `GCC 7.2.1`.
Configure command is `perl tools/builder/build_mfx.pl --no-warn-as-error --prefix=$HOME/Work/workspace/media/install --cmake=intel64.make.release`

```
[ 95%] Building CXX object samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/src/ref_list_manager.cpp.o
cd /home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release/samples/sample_hevc_fei && /usr/bin/c++  -DLINUX -DLINUX32 -DLINUX64 -DLINUX_TARGET_PLATFORM -DLINUX_TARGET_PLATFORM_BDW -DMFX_FILE_VERSION=\"0.18.1.26\" -DMFX_PRODUCT_VERSION=\"0.0.000.0000\" -DMSDK_BUILD=\"\" -DUNIX -D_FILE_OFFSET_BITS=64 -D__USE_LARGEFILE64 -I/home/uaeoff/Work/workspace/media/build/msdk/api/mediasdk_structures -I/home/uaeoff/Work/workspace/media/build/msdk/api/include -I/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/../sample_common/include -I/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/include  -msse4.2 -std=c++11 -m64 -g -pipe -fPIC -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wall -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-unused  -Wformat -Wformat-security -DNDEBUG   -I/usr/include/libdrm -I/home/uaeoff/Work/workspace/media/install/include -I/home/uaeoff/Work/workspace/media/install/include -I/home/uaeoff/Work/workspace/media/install/include  -DMFX_VA -DLIBVA_SUPPORT -DLIBVA_DRM_SUPPORT -DLIBVA_X11_SUPPORT   -I/home/uaeoff/Work/workspace/media/install/include -I/usr/include/libdrm   -DLIBVA_SUPPORT -DLIBVA_DRM_SUPPORT -DLIBVA_X11_SUPPORT    -o CMakeFiles/sample_hevc_fei.dir/src/ref_list_manager.cpp.o -c /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp: In member function ‘bool HevcRplUtils::PocDistanceIsLess::operator()(size_t, size_t) const’:
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:51:49: error: call of overloaded 'abs(unsigned int)' is ambiguous
                 (std::abs(m_dpb[l].m_poc - m_poc)) <
                                                 ^
In file included from /usr/include/c++/7/cstdlib:77:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/c++/7/bits/std_abs.h:78:3: note: candidate: constexpr long double std::abs(long double)
   abs(long double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:74:3: note: candidate: constexpr float std::abs(float)
   abs(float __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:70:3: note: candidate: constexpr double std::abs(double)
   abs(double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:61:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/7/bits/std_abs.h:56:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /usr/include/c++/7/cstdlib:75:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/stdlib.h:722:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:52:49: error: call of overloaded 'abs(unsigned int)' is ambiguous
                 (std::abs(m_dpb[r].m_poc - m_poc));
                                                 ^
In file included from /usr/include/c++/7/cstdlib:77:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/c++/7/bits/std_abs.h:78:3: note: candidate: constexpr long double std::abs(long double)
   abs(long double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:74:3: note: candidate: constexpr float std::abs(float)
   abs(float __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:70:3: note: candidate: constexpr double std::abs(double)
   abs(double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:61:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/7/bits/std_abs.h:56:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /usr/include/c++/7/cstdlib:75:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/stdlib.h:722:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp: In member function ‘bool HevcRplUtils::PocDistanceIsGreater::operator()(size_t, size_t) const’:
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:68:49: error: call of overloaded ‘abs(unsigned int)’ is ambiguous
                 (std::abs(m_dpb[l].m_poc - m_poc)) >
                                                 ^
In file included from /usr/include/c++/7/cstdlib:77:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/c++/7/bits/std_abs.h:78:3: note: candidate: constexpr long double std::abs(long double)
   abs(long double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:74:3: note: candidate: constexpr float std::abs(float)
   abs(float __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:70:3: note: candidate: constexpr double std::abs(double)
   abs(double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:61:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/7/bits/std_abs.h:56:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /usr/include/c++/7/cstdlib:75:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/stdlib.h:722:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:69:49: error: call of overloaded 'abs(unsigned int)' is ambiguous
                 (std::abs(m_dpb[r].m_poc - m_poc));
                                                 ^
In file included from /usr/include/c++/7/cstdlib:77:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/c++/7/bits/std_abs.h:78:3: note: candidate: constexpr long double std::abs(long double)
   abs(long double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:74:3: note: candidate: constexpr float std::abs(float)
   abs(float __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:70:3: note: candidate: constexpr double std::abs(double)
   abs(double __x)
   ^~~
/usr/include/c++/7/bits/std_abs.h:61:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/7/bits/std_abs.h:56:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /usr/include/c++/7/cstdlib:75:0,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/ref_list_manager.cpp:20:
/usr/include/stdlib.h:722:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
make[2]: *** [samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/build.make:210: samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/src/ref_list_manager.cpp.o] Error 1
make[2]: Leaving directory '/home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release'
make[1]: *** [CMakeFiles/Makefile2:1959: samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/all] Error 2
make[1]: Leaving directory '/home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release'
make: *** [Makefile:133: all] Error 2
make: Leaving directory '/home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release'
```

--- a/samples/sample_hevc_fei/src/ref_list_manager.cpp
+++ b/samples/sample_hevc_fei/src/ref_list_manager.cpp
@@ -51,7 +51,7 @@ namespace HevcRplUtils
                 (std::abs(m_dpb[l].m_poc - m_poc)) <
                 (std::abs(m_dpb[r].m_poc - m_poc));
         }
-        mfxU32 m_poc;
+        mfxI32 m_poc;
     };
 
     struct PocDistanceIsGreater : public BasePredicateForRefPicure
@@ -68,7 +68,7 @@ namespace HevcRplUtils
                 (std::abs(m_dpb[l].m_poc - m_poc)) >
                 (std::abs(m_dpb[r].m_poc - m_poc));
         }
-        mfxU32 m_poc;
+        mfxI32 m_poc;
     };
 
     struct InterlacePocDistanceIsLess : public BasePredicateForRefPicure
@@ -89,7 +89,7 @@ namespace HevcRplUtils
                 (std::abs(GetFrameNum(true, m_dpb[r].m_poc, m_dpb[r].m_secondField) - currFrameNum) * 2 + ((m_dpb[r].m_bottomField == m_IsBottomField) ? 0 : 1));
         }
 
-        mfxU32 m_poc;
+        mfxI32 m_poc;
         bool   m_IsSecondField;
         bool   m_IsBottomField;
     };
@@ -112,7 +112,7 @@ namespace HevcRplUtils
                 (std::abs(GetFrameNum(true, m_dpb[r].m_poc, m_dpb[r].m_secondField) - currFrameNum) * 2 + ((m_dpb[r].m_bottomField == m_IsBottomField) ? 0 : 1));
         }
 
-        mfxU32 m_poc;
+        mfxI32 m_poc;
         bool   m_IsSecondField;
         bool   m_IsBottomField;
     };

